diff --git a/pkgs/development/r-modules/default.nix b/pkgs/development/r-modules/default.nix
index 2464b54c5162..0da560c2c43a 100644
--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1073,6 +1073,12 @@ let
       ];
     });
 
+    lwgeom = old.lwgeom.overrideAttrs (attrs: {
+      configureFlags = [
+        "--with-proj-lib=${pkgs.lib.getLib pkgs.proj}/lib"
+      ];
+    });
+
     terra = old.terra.overrideAttrs (attrs: {
       configureFlags = [
         "--with-proj-lib=${pkgs.lib.getLib pkgs.proj}/lib"
