diff --git a/pkgs/development/r-modules/default.nix b/pkgs/development/r-modules/default.nix
index 864ea018b2ee..676ae719e222 100644
--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1796,10 +1796,6 @@ let
       PKGCONFIG_LIBS="-L${pkgs.redland}/lib -L${pkgs.librdf_raptor2}/lib -L${pkgs.librdf_rasqal}/lib -lrdf -lraptor2 -lrasqal";
     });
 
-    textshaping = old.textshaping.overrideAttrs (attrs: {
-      env.NIX_LDFLAGS = "-lfribidi -lharfbuzz";
-    });
-
     httpuv = old.httpuv.overrideAttrs (_: {
       preConfigure = ''
         patchShebangs configure
