diff --git a/pkgs/development/r-modules/default.nix b/pkgs/development/r-modules/default.nix
index 0727d0e6b767..0f9b25afe2eb 100644
--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1672,10 +1672,8 @@ let
       ];
     });
 
-    xslt = old.xslt.overrideAttrs (attrs: {
-      env = (attrs.env or { }) // {
-        NIX_CFLAGS_COMPILE = attrs.env.NIX_CFLAGS_COMPILE + " -fpermissive";
-      };
+    xslt = old.xslt.overrideDerivation (attrs: {
+        NIX_CFLAGS_COMPILE = attrs.NIX_CFLAGS_COMPILE + " -fpermissive";
     });
 
     sparklyr = old.sparklyr.overrideAttrs (attrs: {
