diff --git a/pkgs/development/r-modules/default.nix b/pkgs/development/r-modules/default.nix
index 0f9b25afe2eb..0ebc12a973c4 100644
--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1464,11 +1460,9 @@ let
         '';
     });
 
-    OpenMx = old.OpenMx.overrideAttrs (attrs: {
-      env = (attrs.env or { }) // {
+    OpenMx = old.OpenMx.overrideDerivation (attrs: {
         # needed to avoid "log limit exceeded" on Hydra
-        NIX_CFLAGS_COMPILE = attrs.env.NIX_CFLAGS_COMPILE + " -Wno-ignored-attributes";
-      };
+        NIX_CFLAGS_COMPILE = attrs.NIX_CFLAGS_COMPILE + " -Wno-ignored-attributes";
       preConfigure = ''
         patchShebangs configure
         '';
