diff --git a/pkgs/development/r-modules/default.nix b/pkgs/development/r-modules/default.nix
index 0ebc12a973c4..68ff08998b0c 100644
--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -1480,11 +1480,9 @@ let
         '';
     });
 
-    FlexReg = old.FlexReg.overrideAttrs (attrs: {
-      env = (attrs.env or { }) // {
+    FlexReg = old.FlexReg.overrideDerivation (attrs: {
         # needed to avoid "log limit exceeded" on Hydra
-        NIX_CFLAGS_COMPILE = attrs.env.NIX_CFLAGS_COMPILE + " -Wno-ignored-attributes";
-      };
+        NIX_CFLAGS_COMPILE = attrs.NIX_CFLAGS_COMPILE + " -Wno-ignored-attributes";
 
       # consumes a lot of resources in parallel
       enableParallelBuilding = false;
